### ISSUE TITLE
Vends xhtml files with improper mime type.

### ISSUE THREAD

**Issue by idearat** [phusion/passenger#1695]
Launching a Node.js application which uses xhtml files does not work correctly. The mime type used for xhtml files is incorrect and causes the browser to presume they should be downloaded instead of treating them as renderable resources (XHTML5 content).

I'm running Node.js v0.10.36 and Phusion Passenger version 5.0.7 on Mac OSX 10.10.5 (Yosemite). The Node.js application uses express v.4.13.3 and a simple server.js file I'm starting with the command line:

passenger start --app-type node --startup-file server.js

The current mime.types file (5.0.7/libexec/resources/mime.types contains this entry for XHTML:
    application/vnd.wap.xhtml+xml         xhtml;

However, the IANA registered MIME information shows the entry for XHTML files should be:
    application/xhtml+xml         xhtml;

Refer to these pages for reference:
http://www.iana.org/assignments/media-types/media-types.xhtml
and
http://www.iana.org/assignments/media-types/application/xhtml+xml

Changing the current line to the proposed line repairs this problem and allows the application to render properly rather than downloading XHTML5 page content.

**Comment by OnixGH** [comment]
Are you setting a DOCTYPE in your output?

`application/vnd.wap.xhtml+xml` is the official mime type for xhtml MP (mobile profile), so they are both valid. We could consider setting `application/xhtml+xml` as the new default but I don't know which is more common these days.

**Comment by idearat** [comment]
The documents in question follow the format of this recommendation:

https://blog.whatwg.org/xhtml5-in-a-nutshell

So they all have this skeleton:

<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<title>...</title>
<meta charset="UTF-8" />
</head>

<body>
...
</body>
</html>

**Comment by idearat** [comment]
I'm new to passenger so I have a question now that I think about this. If the underlying Node.js server is vending the files properly (or at least with application/xhtml+xml as desired in this case) why/where in the process would passenger be making the decision to change that to a mobile profile? And if it came down to it shouldn't it be possible to detect whether the incoming request is one that requires a mobile profile?

**Comment by idearat** [comment]
http://caniuse.com/#feat=xhtml

**Comment by OnixGH** [comment]
Can you try this with the newest Passenger (5.0.22)?

If I set the response header `Content-Type` to `application/xhtml+xml` from my Node app, that's how it ends up in my curl output (directly from Node, as well as via Passenger):
```
curl -v http://url/test
..
< HTTP/1.1 200 OK
< Content-Type: application/xhtml+xml
```

**Comment by idearat** [comment]
All xhtml files vend with that content type from the node server itself. Using 5.0.22 didn't change anything in my browser's response. Run my application from Node.js, works fine. Launch and run with Passenger in Standalone mode and all xhtml files fail to vend properly and trigger downloads.

**Comment by OnixGH** [comment]
Aha, sorry, you were talking about statically served files. If you use Passenger/Nginx, then those files are served up by Nginx, which explains why the mime types have an effect (Passenger doesn't touch mime types in application responses).

We tested a few situations, and although Firefox renders it properly (no download), other browsers like Chrome do get a download. We tested a changed version and Android/iOS seem to handle it OK, so we'll change the default.
